File: src/creational/abstract-factory/python/classes/customer_vehicle.py
from classes.interfaces.vehicle_protocol import VehicleProtocol


class CustomerVehicle(VehicleProtocol):
    def pick_up(self, user_name: str) -> None:
        self.customer = user_name
        print(f'{self.get_name()} is looking for {user_name}')

    def get_user(self):
        return self.customer if self.customer else None

File: src/creational/abstract-factory/typescript/classes/CustomerVehicle.ts
import { VehicleProtocol } from './interfaces/VehicleProtocol';

export class CustomerVehicle extends VehicleProtocol {
  private customer: string | null = null;

  pickUp(userName: string): void {
    this.customer = userName;
    console.log(`${this.name} is looking for ${userName}`);
  }

  get user(): string | null {
    return this.customer;
  }
}
